src/image_intel.cpp

```diff
@@ -78,22 +80,45 @@ void IntelPlatformsUpdater::reset()
 
 void IntelPlatformsUpdater::doInstall(const fs::path& file)
 {
-    size_t bootaddr = IMAGE_A_ADDR;
-    const char* mtd = "/dev/mtd/image-a";
+    size_t bootaddr = 0;
+    const char* mtd = nullptr;
+
+    const std::string& filename = file.filename();
 
-    if (getBootAddress() == IMAGE_A_ADDR)
+    if (filename == "image-mtd")
+    {
+        mtd = "/dev/mtd0";
+        releaseFlashDrive();
+    }
+    else if (filename == "image-runtime")
+    {
+        bootaddr = IMAGE_A_ADDR;
+        mtd = "/dev/mtd/image-a";
+
+        if (getBootAddress() == IMAGE_A_ADDR)
+        {
+            bootaddr = IMAGE_B_ADDR;
+            mtd = "/dev/mtd/image-b";
+        }
```

Comment:
I don't insist, but I think that a switch() would look more neat here:
```
switch (getBootAddress())
{
case IMAGE_A_ADDR:
    bootaddr = IMAGE_B_ADDR;
    mtd = "/dev/mtd/image-b";
    break;
case IMAGE_B_ADDR:
    bootaddr = IMAGE_A_ADDR;
    mtd = "/dev/mtd/image-a";
    break;
default:
    // Maybe some error handling here
}
```

What do you think?

Reply:
Probably you are right.
`getBootAddress()` can return zero by some external reasons. This case should be handled as an error.